### ISSUE TITLE
Support circuit for code gen

### ISSUE THREAD

**Issue by its-saddamhowladar** [InsertKoinIO/koin#1869]
Please support circuit for code gen https://github.com/slackhq/circuit

**Comment by arnaudgiuliani** [comment]
Is it binding on Circuit factories?

**Comment by SaddamHowladarGit** [comment]
> Is it binding on Circuit factories?

Yes please support code gen solution to ease boilerplate around generating circuit factories.

Circuit offers a KSP-based code gen solution for Dagger and Anvil or Hilt and those libraries only for jvm based platform AFAIK so can't really work on like IOS etc.

**Comment by arnaudgiuliani** [comment]
any will to inject into circuit?
